src/Pim/Bundle/CatalogBundle/Manager/MediaManager.php

```
@@ -139,7 +139,11 @@ public function handle(ProductMediaInterface $media, $filenamePrefix)
      */
     public function duplicate(ProductMediaInterface $source, ProductMediaInterface $target, $filenamePrefix)
     {
-        $target->setFile(new File($source->getFilePath()));
+        if (null === $path = $this->getFilePath($source)) {
+            return;
```

Yep I agree with that. $this->getFilePath() should return an exception in some cases not only when you don't have READ rights on the file.